File: sample/src/main/java/me/xiaopan/sketchsample/fragment/CircleImageShaperTestFragment.java
package me.xiaopan.sketchsample.fragment;

import android.graphics.Color;
import android.graphics.drawable.BitmapDrawable;
import android.os.Bundle;
import android.view.View;
import android.widget.SeekBar;
import android.widget.TextView;

import me.xiaopan.androidinjector.InjectContentView;
import me.xiaopan.androidinjector.InjectView;
import me.xiaopan.sketch.display.TransitionImageDisplayer;
import me.xiaopan.sketch.drawable.ShapeBitmapDrawable;
import me.xiaopan.sketch.shaper.CircleImageShaper;
import me.xiaopan.sketchsample.MyFragment;
import me.xiaopan.sketchsample.R;
import me.xiaopan.sketchsample.widget.MyImageView;

@InjectContentView(R.layout.fragment_circle_image_shaper)
public class CircleImageShaperTestFragment extends MyFragment {
    @InjectView(R.id.image_circleImageShaperFragment)
    MyImageView imageView;

    @InjectView(R.id.seekBar_circleImageShaperFragment_stroke)
    SeekBar strokeSeekBar;

    @InjectView(R.id.text_circleImageShaperFragment_stroke)
    TextView strokeProgressTextView;

    private int strokeProgress = 5;

    @Override
    public void onViewCreated(View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);


        imageView.getOptions().setImageDisplayer(new TransitionImageDisplayer());

        strokeSeekBar.setMax(100);
        strokeSeekBar.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
            @Override
            public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
                strokeProgressTextView.setText(String.format("%d/%d", progress, 100));
            }

            @Override
            public void onStartTrackingTouch(SeekBar seekBar) {

            }

            @Override
            public void onStopTrackingTouch(SeekBar seekBar) {
                strokeProgress = strokeSeekBar.getProgress();
                apply();
            }
        });
        strokeSeekBar.setProgress(strokeProgress);

        apply();
    }

    private void apply() {
        CircleImageShaper imageShaper = new CircleImageShaper().setStroke(Color.WHITE, strokeProgress);

        BitmapDrawable bitmapDrawable = (BitmapDrawable) getResources().getDrawable(R.drawable.image_test);
        ShapeBitmapDrawable shapeBitmapDrawable = new ShapeBitmapDrawable(bitmapDrawable, imageShaper);
        imageView.setImageDrawable(shapeBitmapDrawable);
    }
}

File: sample/src/main/java/me/xiaopan/sketchsample/fragment/CircleImageShaperTestFragment.kt
package me.xiaopan.sketchsample.fragment

import android.graphics.Color
import android.os.Bundle
import android.view.View
import android.widget.SeekBar
import android.widget.TextView
import me.xiaopan.sketch.display.TransitionImageDisplayer
import me.xiaopan.sketch.shaper.CircleImageShaper
import me.xiaopan.sketchsample.AssetImage
import me.xiaopan.sketchsample.BaseFragment
import me.xiaopan.sketchsample.BindContentView
import me.xiaopan.sketchsample.R
import me.xiaopan.sketchsample.widget.SampleImageView
import me.xiaopan.ssvt.bindView

@BindContentView(R.layout.fragment_circle_image_shaper)
class CircleImageShaperTestFragment : BaseFragment() {
    val imageView: SampleImageView by bindView(R.id.image_circleImageShaperFragment)
    val strokeSeekBar: SeekBar by bindView(R.id.seekBar_circleImageShaperFragment_stroke)
    val strokeProgressTextView: TextView by bindView(R.id.text_circleImageShaperFragment_stroke)

    private var strokeProgress = 5

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)


        imageView.options.displayer = TransitionImageDisplayer()

        strokeSeekBar.max = 100
        strokeSeekBar.setOnSeekBarChangeListener(object : SeekBar.OnSeekBarChangeListener {
            override fun onProgressChanged(seekBar: SeekBar, progress: Int, fromUser: Boolean) {
                strokeProgressTextView.text = String.format("%d/%d", progress, 100)
            }

            override fun onStartTrackingTouch(seekBar: SeekBar) {

            }

            override fun onStopTrackingTouch(seekBar: SeekBar) {
                strokeProgress = strokeSeekBar.progress
                apply()
            }
        })
        strokeSeekBar.progress = strokeProgress

        apply()
    }

    private fun apply() {
        val imageShaper = CircleImageShaper().setStroke(Color.WHITE, strokeProgress)

        imageView.options.shaper = imageShaper
        imageView.displayImage(AssetImage.TYPE_TEST_JPG)
    }
}